crypto/bindings/util.ts

```diff
@@ -21,3 +21,16 @@ function mapTuple<T extends Tuple<any>, B>(
 ): { [i in keyof T]: B } {
   return tuple.map(f) as any;
 }
+
+/**
+ * tuple type that has the length as generic parameter
+ */
+type MlTupleN<T, N extends number> = N extends N
```

Comment:
:exploding_head:

Reply:
I don't understand this. What is supposed to represent?

Reply:
let's say you want a type for a static size tuple like `field * field * field * field * field`.
The TS version would be `[0, Field, Field, Field, Field, Field]`

this generic type let's you instead write `MlTuple<Field, 5>` so it saves you some ugliness (especially in the case of size-15 tuples that we have in some places)

Reply:
also let's you have functions like `mapTuple` that are generic over the tuple length but still preserve its typed size